stagecoach/price-pusher/price_pusher/orchestrator.py

```diff
@@ -0,0 +1,159 @@
+        is pushed.
+        """
+        await asyncio.gather(listener.run_forever(), self._handle_listener(listener))
+
+    async def _handle_listener(self, listener: PriceListener) -> None:
+        """
+        Monitor a listener in its own thread - wait for it to push an event, and
+        if an event occurs, clear it and retrieves the entries that it was
+        listening and push them to the push queue.
+        """
+        while True:
+            await listener.notification_event.wait()
+            assets_to_push = listener.price_config.get_all_assets()
+            logger.info(
+                f"💡 Notification received from LISTENER [{listener.id}] ! "
+                "Sending entries into queue for: "
+                f"{[asset_to_pair_id(asset) for asset in assets_to_push]}"
+            )
+            entries_to_push = self._flush_entries_for_assets(assets_to_push)
+            await self.push_queue.put(entries_to_push)
+            listener.notification_event.clear()
+
+    async def _pusher_service(self) -> None:
+        """
+        Service reponsible of publishing entries that are in the queue.
+        """
+        while True:
+            entries_to_push = await self.push_queue.get()
+            await self.pusher.update_price_feeds(entries_to_push)
+            self.push_queue.task_done()
+
+    def _flush_entries_for_assets(self, assets: List[PragmaAsset]) -> List[Entry]:
+        """
+        Retrieves the prices for the assets that needs to be pushed & remove them from
+        the latest_prices dict.
+        """
+        entries_to_push = []
+        for asset in assets:
+            pair_id = asset_to_pair_id(asset)
+            if pair_id in self.latest_prices:
+                for entry_type, sources in self.latest_prices[pair_id].items():
+                    for source, entry in sources.items():
+                        entries_to_push.append(entry)
+                del self.latest_prices[pair_id]
+        return entries_to_push
+
+    def callback_update_prices(self, entries: List[Entry]) -> None:
+        """
+        Function called by the poller whenever new prices are retrieved.
+        """
+        for entry in entries:
+            pair_id = entry.get_pair_id()
+            source = entry.get_source()
+            entry_type: Optional[AssetType] = (
+                "SPOT"
+                if isinstance(entry, SpotEntry)
+                else "FUTURE"
+                if isinstance(entry, FutureEntry)
+                else None
+            )
+            if entry_type is None:
+                logger.error(f'Entry type of "{pair_id}" from "{source}" is unknown. Ignoring.')
+                continue
+
+            if pair_id not in self.latest_prices:
+                self.latest_prices[pair_id] = {}
+            if entry_type not in self.latest_prices[pair_id]:
+                self.latest_prices[pair_id][entry_type] = {}
```

Comment:
what's the point of storing `entry_type` and `source` in this mapping since they are not used at all?

Reply:
we're using them to make sure that for all those components we always have on entry, makes it easy to check & replace that

what would be another solution?